src/ci/docker/x86_64-gnu-tools/repo.sh

```diff
@@ -62,6 +62,10 @@ commit_toolstate_change() {
     MESSAGE_FILE="$1"
     shift
     for RETRY_COUNT in 1 2 3 4 5; do
+        # Call the callback; this will in the end call `change_toolstate` from
+        # `checktools.sh` if we are in the `auto` branch (pre-landing) or
+        # `src/tools/publish_toolstate.py` if we are in the `master` branch
```

Comment:
Is it correct to say that `publish_toolstate` does *not* publish the toolstate, but all it actually does is put a comment on the PR that caused this, and open an issue to track the failed tool?

Reply:
Hm, it also puts a message into `$(mktemp -t msg.XXXXXX)`. Not sure if that is used for anything though.

it does not at all interact with the https://github.com/rust-lang-nursery/rust-toolstate repo, from what I can see.

Reply:
Ah, that message file is used as *commit message* for the toolstate repo. It ends up creating commits like this https://github.com/rust-lang-nursery/rust-toolstate/commit/689516a5d57571f3a2d2c4198b187039ed6f83c8. So it does "publish the toolstate" in some sense.

The confusion is between publishing the test results and publishing the toolstate, which are separate things. And also between there being very few comments explaining all that. ;)